library/core/src/main/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelector.java

```diff
@@ -1552,38 +1552,30 @@ public void experimental_allowMultipleAdaptiveSelections() {
       }
     }
 
-    int selectedTextTrackScore = Integer.MIN_VALUE;
+    TextTrackScore selectedTextTrackScore = null;
     int selectedTextRendererIndex = C.INDEX_UNSET;
     for (int i = 0; i < rendererCount; i++) {
-      int trackType = mappedTrackInfo.getRendererType(i);
-      switch (trackType) {
-        case C.TRACK_TYPE_VIDEO:
-        case C.TRACK_TYPE_AUDIO:
-          // Already done. Do nothing.
-          break;
-        case C.TRACK_TYPE_TEXT:
-          Pair<TrackSelection.Definition, Integer> textSelection =
-              selectTextTrack(
-                  mappedTrackInfo.getTrackGroups(i),
-                  rendererFormatSupports[i],
-                  params,
-                  selectedAudioLanguage);
-          if (textSelection != null && textSelection.second > selectedTextTrackScore) {
-            if (selectedTextRendererIndex != C.INDEX_UNSET) {
-              // We've already made a selection for another text renderer, but it had a lower score.
-              // Clear the selection for that renderer.
-              definitions[selectedTextRendererIndex] = null;
-            }
-            definitions[i] = textSelection.first;
-            selectedTextTrackScore = textSelection.second;
-            selectedTextRendererIndex = i;
+      // The below behaviour is different from video and audio track selection
+      // i.e. do not perform a text track pre selection if there are no preferredTextLanguage requested.
+      if (C.TRACK_TYPE_TEXT == mappedTrackInfo.getRendererType(i) && params.preferredTextLanguage != null) {
+        Pair<TrackSelection.Definition, TextTrackScore> textSelection =
+            selectTextTrack(
+                mappedTrackInfo.getTrackGroups(i),
+                rendererFormatSupports[i],
+                params);
+        if (textSelection != null
+            && (selectedTextTrackScore == null
+            || textSelection.second.compareTo(selectedTextTrackScore) > 0)) {
+          if (selectedTextRendererIndex != C.INDEX_UNSET) {
+            // We've already made a selection for another text renderer, but it had a lower
+            // score. Clear the selection for that renderer.
+            definitions[selectedTextRendererIndex] = null;
           }
-          break;
-        default:
-          definitions[i] =
-              selectOtherTrack(
```

Comment:
You can't remove that. We still need to select "other" tracks.

Reply:
Thanks for your comment, I re-added it in the next commit.